server/src/main/java/org/elasticsearch/plugins/Plugin.java

```diff
@@ -205,4 +206,9 @@ public void close() throws IOException {
     public Collection<IndexSettingProvider> getAdditionalIndexSettingProviders(IndexSettingProvider.Parameters parameters) {
         return Collections.emptyList();
     }
+
+    /**
+     * Called with a Tracer so that each plugin has a chance to work with it.
+     */
+    public void onTracer(Tracer tracer) {}
```

Comment:
"onXXX" doesn't really fit with other plugin methods (the one exception is onIndexModule, but that is a holdover from the older style push based model, which index specific plugin extensions still use). The purpose of this seems to be to give plugins this global Tracer object, so that they can stash it to use later.

If the tracer object needs to be passed around the entire system, then it doesn't seem any better than just having actual global state for this purpose. Otherwise every time we realize something else needs access to the tracer, we have to plumb it through, which it seems like could eventually be the entire system? If I've misread the purpose of this object, then it would seem better as an argument to createComponents, where other "common" systems are passed to plugins so they can be utilized in plugin service construction.